src/internal/grid/grid.tsx

```diff
@@ -3,38 +3,52 @@
 
 import { useContainerQuery } from "@cloudscape-design/component-toolkit";
 import clsx from "clsx";
-import { Children } from "react";
+import { Children, useRef } from "react";
+import { useMergeRefs } from "../utils/use-merge-refs";
 import { zipTwoArrays } from "../utils/zip-arrays";
 
 import { GridProps } from "./interfaces";
 import GridItem from "./item";
 import styles from "./styles.css.js";
+import { useDensityMode } from "./use-density-mode";
```

Comment:
Are we short on time for this feature? How about doing it properly and bubble up this utility into the toolkit?

Reply:
I would do this as a follow-up for a number of reasons. First, I am not sure if it rather belongs to the toolkit or global styles. Second, the mutation-observer based detection is already known for its impact on performance and it might make sense to not share it publicly. We can even consider using a simpler detection for board-components, e.g. by monitoring the body only.

Reply:
I meant the internal folder: https://github.com/cloudscape-design/component-toolkit/tree/main/src/internal

Reply:
Yes, I assumed the internal scope but still it makes the tools more exposed. However, I am not strictly against that. Let's discuss here: https://github.com/cloudscape-design/component-toolkit/pull/29